lib/grpcServer/handlers/platform/getDocumentsHandlerFactory.js

```diff
@@ -0,0 +1,87 @@
+const cbor = require('cbor');
+
+const {
+  server: {
+    error: {
+      InvalidArgumentGrpcError,
+    },
+  },
+} = require('@dashevo/grpc-common');
+
+const {
+  GetDocumentsResponse,
+} = require('@dashevo/dapi-grpc');
+
+/**
+ *
+ * @param {DriveAdapter} driveAPI
+ * @param {DashPlatformProtocol} dpp
+ * @returns {getDocumentsHandler}
+ */
+function getDocumentsHandlerFactory(driveAPI, dpp) {
+  /**
+   * @typedef getDocumentsHandler
+   * @param {Object} call
+   * @returns {Promise<GetDocumentsResponse>}
+   */
+  async function getDocumentsHandler(call) {
+    const { request } = call;
+
+    const dataContractId = request.getDataContractId();
+
+    if (!dataContractId) {
+      throw new InvalidArgumentGrpcError('dataContractId is not specified');
+    }
+
+    const documentType = request.getDocumentType();
+
+    if (!documentType) {
+      throw new InvalidArgumentGrpcError('documentType is not specified');
+    }
+
+    const whereBinary = request.getWhere();
+
+    let where;
+    if (whereBinary) {
+      where = cbor.decode(whereBinary);
+    }
+
+    const orderByBinary = request.getOrderBy();
+
+    let orderBy;
+    if (orderByBinary) {
+      orderBy = cbor.decode(orderByBinary);
+    }
+
+    const limit = request.getLimit();
+
+    const startAfter = request.getStartAfter();
+    const startAt = request.getStartAt();
+
+    const options = {
+      where,
+      orderBy,
+      limit,
+      startAfter,
+      startAt,
+    };
+
+    const documentsJSON = await driveAPI.fetchDocuments(dataContractId, documentType, options);
```

Comment:
It might throw an invalid argument grpc error which is just `Error` so we need to catch it and wrap into `InvalidArgumentGrpcError` as we are doing in https://github.com/dashevo/js-machine/blob/v0.3-dev/lib/abci/handlers/deliverTxHandlerFactory.js#L114